### PR TITLE
GraphQL v2 Tier updates

### PR DESCRIPTION
**Pull request by sbinlondon**
* Adds more fields to the Tier object to be used on the frontend
* Adds a TierCollection
* Allows you to query Tiers on a Collective in v2
* Updates the `updateOrder` mutation to use the new v2 stuff passed from the frontend

### CODE REVIEW

File: server/graphql/v2/object/Tier.js [hunk 1]
```diff
@@ -62,6 +64,36 @@ export const Tier = new GraphQLObjectType({
...
         },
       },
+      amount: {
+        type: Amount,
+        resolve(tier) {
+          return { value: tier.amount, currency: tier.currency };
+        },
+      },
+      interval: {
+        type: GraphQLString,
```
Comment (Betree) [inline server/graphql/v2/object/Tier.js:74]: **[:sparkles: **Enhancement**]** Would be nice to create an enum for this one

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

File: server/graphql/v2/object/Tier.js [hunk 2]
```diff
@@ -62,6 +64,36 @@ export const Tier = new GraphQLObjectType({
...
+        },
+      },
+      presets: {
+        type: new GraphQLList(GraphQLInt),
+        resolve(tier) {
+          return tier.presets;
+        },
+      },
+      amountType: {
+        type: GraphQLString,
```
Comment (Betree) [inline server/graphql/v2/object/Tier.js:86]: - Should be not nullable
- Would be nice to create an enum for this one

```suggestion
        type: new GraphQLNonNull(GraphQLString),
```

File: server/graphql/v2/input/AmountInput.js [hunk 3]
```diff
@@ -13,5 +14,9 @@ export const AmountInput = new GraphQLObjectType({
       type: Currency,
       description: 'The currency string',
     },
+    valueInCents: {
+      type: GraphQLFloat,
```
Comment (znarf) [inline server/graphql/v2/input/AmountInput.js:18]: Should normally be an Int

File: server/graphql/v2/object/Tier.js (hunk 1 above)
Comment (sbinlondon) [reply server/graphql/v2/object/Tier.js:74]: Oh you're right, I think there might actually be one already 🤔

File: server/graphql/v2/input/TierReferenceInput.js [hunk 4]
```diff
@@ -1,11 +1,38 @@
...
   }),
 });
+
+/**
+ * Retrieves a tier
+ *
+ * @param {string|number} input - id of the tier
+ */
+export const fetchTierWithReference = async input => {
+  const loadTierById = id => {
```
Comment (znarf) [inline server/graphql/v2/input/TierReferenceInput.js:23]: Looks like it's not needed, you can simply use `models.Tier.findByPk` later.

Comment (sbinlondon) [reply server/graphql/v2/input/TierReferenceInput.js:23]: Why would we use `models.Tier.findByPk` in this case but `fetch(Account/PaymentMethod)WithReference` in other places?

Comment (znarf) [reply server/graphql/v2/input/TierReferenceInput.js:23]: I'm writing about `loadTierById` which looks like an useless abstraction.

File: server/graphql/v2/mutation/OrderMutations.js [hunk 5]
```diff
@@ -184,41 +183,40 @@ const orderMutations = {
...
               `This tier (#${tierInfo.id}) doesn't belong to the given Collective #${order.CollectiveId}`,
             );
           }
         }
-        // check if the tier is different from the previous tier
-        if (tier.legacyId !== order.TierId) {
-          order = await order.update({ TierId: tier.legacyId });
+
+        // The amount can never be less than $1.00
+        if (args.amount.valueInCents < 100) {
```
[the remark this answers is not in the document]
Comment (sbinlondon) [reply server/graphql/v2/mutation/OrderMutations.js:203]: There already seems to be a math lib helper called `floatAmountToCents` to I will use that instead.

File: server/graphql/v2/mutation/OrderMutations.js [hunk 6]
```diff
@@ -184,41 +183,40 @@ const orderMutations = {
...
+          throw new Error('Invalid amount.');
+        }
+
+        // If using a named tier, amount can never be less than the minimum amount
+        if (tierInfo && tierInfo.amountType === 'FLEXIBLE' && args.amount.valueInCents < tierInfo.minimumAmount) {
```
[the remark this answers is not in the document]
Comment (sbinlondon) [reply server/graphql/v2/mutation/OrderMutations.js:208]: I think this is a good idea and I will write it down for enhancement in a separate PR. Right now I think it's good to get the feature out for testing.

File: server/graphql/v2/mutation/OrderMutations.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Betree) [reply server/graphql/v2/mutation/OrderMutations.js]: It's not the same thing, my feedback was not about converting the value to cents but about the `AmountInput` object not being handled correctly. If we only use `amount.value` and ignore `amount.valueInCents` then there's no point in adding this field.

Let's say I'm an OC user willing to implement a feature for this endpoint and I decide to use the `valueInCents` field, my code will not work and I'll be surprised. We've introduced a bug here.

**Pull request merged**